pkg/gofr/http/form_data_binder.go

```diff
@@ -0,0 +1,211 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+func (*formData) setInterfaceValue(value reflect.Value, data any) (bool, error) {
+	if !value.CanSet() {
+		return false, fmt.Errorf("%w: %s", errUnsupportedInterfaceType, value.Kind())
+	}
+
+	value.Set(reflect.ValueOf(data))
+
+	return true, nil
+}
+
+func (uf *formData) setSliceOrArrayValue(value reflect.Value, data string) (bool, error) {
+	elemType := value.Type().Elem()
+
+	elements := strings.Split(data, ",")
+
+	// Create a new slice/array with appropriate length and capacity
+	var newSlice reflect.Value
+
+	switch value.Kind() {
+	case reflect.Slice:
+		newSlice = reflect.MakeSlice(value.Type(), len(elements), len(elements))
+	case reflect.Array:
+		if len(elements) > value.Len() {
+			return false, errDataLengthExceeded
+		}
+
+		newSlice = reflect.New(value.Type()).Elem()
+	case reflect.Invalid, reflect.Bool, reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Uint,
```

Comment:
The case seems un-necessary since it is already checked before that if value.Kind is either slice or array

Reply:
Sure i have replaced the `switch` here with if-else